packages/apple-pay-integration/src/apple-pay-button-strategy.ts

```diff
@@ -103,24 +114,63 @@ export default class ApplePayButtonStrategy implements CheckoutButtonStrategy {
         return button;
     }
 
-    private _handleWalletButtonClick(event: Event) {
+    private async _handleWalletButtonClick(event: Event) {
         event.preventDefault();
+        if (
+            this._buyNowInitializeOptions &&
+            typeof this._buyNowInitializeOptions.getBuyNowCartRequestBody === 'function'
+        ) {
+            if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
```

Comment:
This check can be moved outside of parent if condition because it used in both cases.
please move this code before line 119 and remove it the same in else condition.